scenarios.md

```diff
@@ -0,0 +1,190 @@
+
+### Scenario #7 Automated Deploy - Success
+
+A CD solution initiates a deployment to a development environment. The development environment has been previously configured to only accept signed artifacts.
+
+1. A CD solution is triggered to initiate a deployment
+1. The CD solution uses a deployment template to request an orchestrator to run specific images.
+1. The orchestrator receives the request, validates the artifact is signed and the signature is valid *before* it can initiate the deployment.
+1. The signature is considered valid, and the deployment is successful.
+
+**Implications of this requirement:**
+
+- The aspect of being a signed artifact is not germane to the CD solution. The CD solution simply requests a deployment. The enforcement of signature validation is a policy of the execution environment.
+
+### Scenario #8 Automated Deploy - Failure
+
+Same as #7, with the exception that the signature is deemed invalid.
+1. A CD solution is triggered to initiate a deployment.
+2. The CD solution uses a deployment template to request an orchestrator to run a specific image. In this case, the CD solution mistakenly attempted to deploy a dev signed artifact to a production environment.
+3. The production orchestrator receives the request, validates the artifact is signed and recognizes the signature is **NOT** one of the production accepted signatures. As the signature is deemed invalid, the deployment fails.
+
+**Implications of this requirement:**
+- The coordination between a CD and operational environment for how a request and response is managed is not in-scope for artifact signing.
+- An operational environment enforces a policy for which signatures it accepts. Although the artifact signature may be valid, it's NOT the signature required by this operational environment.
+
+### Scenario #9: Multiple Signatures
+
+Customers may require multiple signatures for the following scenarios:
+
+- Validate the artifact is the same as what the vendor provided.
+- Secondarily sign the artifact by the consuming company, attesting to it's validity within their production environment.
+- Signatures represent validations through different dev, staging, production environments.
+
+1. A CI/CD solution builds, signs, pushes and deploys a collection of artifacts to a staging environment.
+1. Once integrations tests are completed, the artifacts are signed with a production signature, copying them to a production registry or production set of repositories.
+1. The Staging and Production container hosts validate the artifacts contain the signatures required for their environment.
+
+**Implications of this requirement:**
+
+- Multiple signatures, including signatures from multiple sources can be associated with a specific artifact.
+
```

Comment:
One of the scenarios that we talked about in the face-to-face meeting but isn't covered here is how signatures are expected to work with base images/layer inheritance.  For example, consider the following:

An image A consists of layers 1, 2, and 3 and is signed by vendor FOO.  This image is then used as the basis of image B which adds layers 4, 5, and 6 and is signed by another party BAR.

Can we validate that layers 1, 2, and 3 (the base image) still have a valid signature by FOO?  Or are we only able to validate the signature of all the layers generated by BAR?

Reply:
I was hoping to account for this. In the Hackmd.io history of notes you can find the following:
> ### Inherited Layer Signatures
> Should signature verification support artifacts that source from another? The most obvious are docker images, where one image is `FROM`-d another. Should a user be able to see the inheritance of an artifact, and know the signature of the base image, the layered runtime are represented in the artifact in question? 

There was a comment that this wasn't feasible. I don't know whether we should pull it in as a scenario, discuss and account for, defer, or promote this to be an SBOM scenario where we can see the history of an image. 

I do believe there's a common request for understanding what base layers are used. Suggesting we defer and add later. 
Feedback?

Reply:
> There was a comment that this wasn't feasible.

I think I missed that part of the discussion.  If I read into it a bit, someone might have thought this to be infeasible as the manifest is the actual part that is signed and a derived image will have a different manifest (and typically a different config).  But we might be able to save the original manifest (and config) as well and use that just for signature validation if we really want to cover this.

I think it'd be worth including this as a scenario and see if we can come up with an approach for it that doesn't have too many downsides.  Anyone else have an opinion here?

Reply:
I do like the scenario, but I don't yet know how to think about it as it pertains to the SBOM discussions that *might be* a better tool. 
I'd suggest the following: What if we add it after the base set are in. This way we can discuss that specific addition, while the common/agreed upon can be in for the spec to begin.

Reply:
I opened #4 to capture this, allowing us to resolve this part of the discussion for this pr.